zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -418,11 +421,10 @@ where
 /// * `prover`: The TxProver to use in constructing the shielded transaction.
 /// * `sk`: The secp256k1 secret key that will be used to detect and spend transparent
 ///   UTXOs.
-/// * `extfvk`: The extended full viewing key that will be used to produce the
-///   Sapling address to which funds will be sent.
-/// * `account`: The ZIP32 account identifier associated with the the extended
-///   full viewing key. This procedure will return an error if this does not correctly
-///   correspond to `extfvk`.
+/// * `account`: The ZIP32 account identifier for the account to which funds will
+///   be shielded. Funds will be shielded to the internal (change) address associated with the
```

Comment:
Hmm; is it intended that `shield_transparent_funds` can shield funds from an _arbitrary_ account (specifically whichever account `sk` corresponds to) to `account`? Or should it only be shielding funds within `account` itself?

Reply:
If we want to enforce that constraint, we will just have to check that correspondence and return an error if it doesn't hold.  Opened https://github.com/zcash/librustzcash/issues/616 to track this.